---
title: 10.2 异常处理机制 10.2.1 使用try-catch捕获异常
categories: 
  - 1 疯狂Java讲义 (第4版)
  - 第10章 异常处理
  - 10.2 异常处理机制
abbrlink: 1cd07df2
date: 2019-07-12 21:56:26
updated: 2022-04-03 01:21:14
---
# 10.2 异常处理机制 #
`Java`的异常处理机制可以让程序具有极好的容错性,让程序更加健壮。**当程序运行出现意外情形时,系统会自动生成一个`Exception`对象来通知程序**,从而实现**将"业务功能实现代码"和"错误处理代码"分离**,提供更好的可读性。
# 10.2.1 使用try...catch捕获异常 #
下面是`Java`异常处理机制的语法结构:
```java
try
{
    //业务实现代码
    ...
}
catch (Exception e)
{
    //错误处理代码
    ...
}
```
## 抛出异常的过程 ##
如果执行`try`块里的业务逻辑代码时出现异常,系统自动生成一个异常对象,该异常对象被提交给**`Java`运行时环境**,这个过程被称为抛出(`throw`)异常。
## 捕获异常的过程 ##
当`Java`运行时环境收到异常对象时,会寻找能处理该异常对象的`catch`块,
如果找到合适的`catch`块,则把该异常对象交给该`catch`块处理,这个过程被称为捕获(`catch`)异常;
如果`Java`运行时环境找不到捕获异常的`catch`块,则运行时环境终止,`Java`程序也将退出。
## 出现异常时没有处理该异常的catch块时程序直接退出 ##
不管程序代码块是否处于`try`块中,甚至包括`catch`块中的代码,**只要执行该代码块时出现了异常,系统总会自动生成一个异常对象**。如果程序没有为这段代码定义任何的`catch`块,则`Java`运行时环境无法找到处理该异常的`catch`块,程序就在此退出。

